### PR TITLE
Refine site search based on user feedback

### PR DESCRIPTION
**Pull request by cpeel**
Refine the new site search experience after input from users. In brief, this:
* Removes the "jump to username" feature if the string is a username, now this needs to be prefixed with `u:`.
* Adds an icon to the search menu to get to the site search help page
* If no prefix was used the search page now lists suggestions to help the user (thanks @windymilla for this suggestion)
* Enables searching for a task by summary in addition to string (thanks @srjfoo for the suggestion)

Sandbox: https://www.pgdp.org/~cpeel/c.branch/refine-site-search/

Forum discussion: https://www.pgdp.net/phpBB3/viewtopic.php?t=81458

### CODE REVIEW

**Review by srjfoo — APPROVED** [review]
FWIW, the task search actually searches in both the summary and the details, not just the summary. I think the description is fine. It's already the longest one :D), but thought I'd clarify that.

**Review by windymilla — APPROVED** [review]
LGTM!

**Review by chrismiceli — APPROVED** [review]

File: tools/site_search.php [hunk 1]
```diff
@@ -44,6 +42,17 @@
 echo "<input type='submit' value='" . attr_safe(_("Search")) . "'>";
 echo "</form>";
 
+if ($query && $query != "help") {
+    echo "<p class='warning'>" . sprintf(_("Not sure what to do with '%s', perhaps try one of these."), $query) . "</p>";
```
Comment (bpfoley) [inline tools/site_search.php:46]: I think we want to `html_safe($query)` here and below

**Review by bpfoley — APPROVED** [review]

File: tools/site_search.php (hunk 1 above)
Comment (cpeel) [reply tools/site_search.php]: Oh yes - good catch! Done and squashed into the relevant (last) commit in the branch.

**Review by 70ray — APPROVED** [review]

**Pull request merged**